src/meta/src/barrier/mod.rs

```diff
@@ -846,12 +853,26 @@ impl GlobalBarrierManager {
                     );
 
                     // This RPC returns only if this worker node has injected this barrier.
-                    client.inject_barrier(request).await
+                    client
+                        .inject_barrier(request)
+                        .await
+                        .map_err(|e| (*node_id, e))
```

Comment:
What about `zip` the results with `info.node_map.iter()` to get the node id **there**?